src/software/gui/drawing/obstacle_artist.cpp

```diff
@@ -1,21 +1,53 @@
 #include "software/gui/drawing/obstacle_artist.h"
 
+#include "proto/geometry.pb.h"
+#include "proto/message_translation/tbots_geometry.h"
+#include "software/logger/logger.h"
+
 ObstacleArtist::ObstacleArtist(QGraphicsScene* scene, const QPen& pen)
     : scene_(scene), pen_(pen)
 {
 }
 
 void ObstacleArtist::visit(const GeomObstacle<Circle>& geom_obstacle)
 {
+    TbotsProto::Circle circle_proto;
+    auto circle = geom_obstacle.getGeom();
+
+    *(circle_proto.mutable_origin()) = *createPointProto(circle.origin());
+    circle_proto.set_radius(circle.radius());
+    *(obstacle_proto_.add_circle()) = circle_proto;
+
     drawCircle(scene_, geom_obstacle.getGeom(), pen_);
 }
 
 void ObstacleArtist::visit(const GeomObstacle<Polygon>& geom_obstacle)
 {
+    TbotsProto::Polygon poly_proto;
+    std::vector<Point> points = geom_obstacle.getGeom().getPoints();
+    for (const auto& point : points)
+    {
+        *(poly_proto.add_points()) = *createPointProto(point);
+    }
```

Comment:
We should probably add a `Polygon` -> `TbotsProto::Polygon` translation function to avoid duplicated translation code.

Reply:
good catch! changed to use translation functions, looks nicer now